File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/overview/sort_mode_header/RecyclerItemsHeaderStrategy.java
package coolpharaoh.tee.speicher.tea.timer.views.overview.sort_mode_header;

import java.util.List;

import coolpharaoh.tee.speicher.tea.timer.core.tea.Tea;
import coolpharaoh.tee.speicher.tea.timer.views.overview.RecyclerItemOverview;

public interface RecyclerItemsHeaderStrategy {

    public List<RecyclerItemOverview> generateFrom(final List<Tea> teaList);
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/overview/recycler_view/RecyclerItemsHeaderStrategy.kt
package coolpharaoh.tee.speicher.tea.timer.views.overview.recycler_view

import coolpharaoh.tee.speicher.tea.timer.core.tea.Tea

interface RecyclerItemsHeaderStrategy {
    fun generateFrom(teaList: List<Tea>): List<RecyclerItemOverview>
}